### PR TITLE
update route-controller-manager and wire context correctly

### PR DESCRIPTION
**Pull request by atiratree**
built on top of https://github.com/openshift/microshift/pull/1032

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/openshift/microshift
 go 1.17
 
 require (
-	github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e // route-controller-manager
+	github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e // microshift
```
Comment (atiratree) [inline go.mod:6]: rcm lost this dependency https://github.com/openshift/route-controller-manager/pull/9, so now it is just a direct dependency of microshift

File: pkg/controllers/openshift-route-controller-manager.go [hunk 2]
```diff
@@ -116,9 +116,5 @@ func (s *OCPRouteControllerManager) Run(ctx context.Context, ready chan<- struct
 		return err
 	}
 
-	if err := route_controller_manager.RunRouteControllerManager(s.config, clientConfig); err != nil {
-		return err
-	}
-	<-ctx.Done()
-	return nil
+	return route_controller_manager.RunRouteControllerManager(s.config, clientConfig, ctx)
```
Comment (atiratree) [inline pkg/controllers/openshift-route-controller-manager.go:119]: reacting to https://github.com/openshift/route-controller-manager/pull/6 change

**Comment by dhellmann** [comment]
part of WRKLDS-487

**Comment by atiratree** [comment]
done in #1055

**Pull request closed without merging**